src/posix/platform/radio.hpp

```diff
@@ -113,4 +122,5 @@ class Radio
 } // namespace Posix
 } // namespace ot
 
+ot::Posix::Radio *GetRadio(void);
```

Comment:
Do we need this?

Reply:
VendorRadioSpinel look something like:
```
class VendorRadioSpinel : public ot::Spinel::RadioSpinel
{
public:
    VendorRadioSpinel(void)
        : ot::Spinel::RadioSpinel()
    {
    }

    otError DoSomething(void);
    // Add public/private methods or member variables
};
```
We need access to `ot::Posix::Radio` in order to do something like the below from a registered callback
```
ot::Spinel::VendorRadioSpinel* spinel = (ot::Spinel::VendorRadioSpinel*)(&GetRadioSpinel());
error = spinel->DoSomething();
VerifyOrDie((error == OT_ERROR_NONE) || (error == OT_ERROR_NOT_IMPLEMENTED), OT_ERROR_FAILED);
```

Reply:
@abtink: unless you have another suggestion

Reply:
Not sure if I follow, where would we use the `Posix::Radio` instance?

Reply:
I mixed up `VendorRadioSpinel` and `Posix::Radio`. I have removed it